### ISSUE TITLE
Next steps for library

### ISSUE THREAD

**Issue by straker** [straker/kontra#118]
Twitter Poll: https://twitter.com/StevenKLambert/status/1172875307767889927
Slack Poll: https://js13kgames.slack.com/archives/C1EHVV2DN/p1568471866031800

Overall result:
- More modular to allow for more code elimination (things like rotation in Sprite or '1..10' in SpriteSheet) and allow for more features

**Comment by crhallberg** [comment]
What do you think of the idea of adding a new build task that builds each src to its own dest. That would make it easy to pick and choose which parts you use.

```html
<script src="kontra/core.js"></script>
<script src="kontra/gameLoop.js"></script>
<script src="kontra/sprite.js"></script>
<script src="kontra/animation.js"></script>
```

I don't have a solution for certain dependencies. For example, GameLoop can work without  `getContext` but it loses the default clearFn. However, Sprite basically requires `getContext` to work. Surrounding everything with like `if (typeof kontra.emit !== "undefined")` is going to balloon the file size for this kind of flexibility.

**Comment by straker** [comment]
So the previous version of the library had that so you could do builds that way, but with the library having been converted to ES6, there's no need anymore. The recommended way to do custom builds to get the parts you need is to use Rollup or Webpack which also gets you tree shaking.

**Comment by wini3d** [comment]
I agree with the comments there about more features through plugins. Also, would be nice to have a solid collision system and cover the most used methods by examples.

**Comment by straker** [comment]
Closed via #169
